core/src/main/resources/error/error-classes.json

```diff
@@ -933,6 +933,11 @@
     ],
     "sqlState" : "42000"
   },
+  "TEMP_VIEW_DOES_NOT_BELONG_TO_A_DATABASE" : {
```

Comment:
Since we are touching this error, let's take the chance to refine it. How about
```
"CREATE_TEMP_VIEW_SINGLE_PART_NAME": {
  "message": [
    "CREATE TEMPORARY VIEW or the corresponding Dataset APIs only accept single-part view names, but got: <actualName>"
  ]
}
```

Reply:
cc @MaxGekk

Reply:
By doing this, we don't need https://github.com/apache/spark/pull/38595/files#diff-459628811d7786c705fbb2b7a381ecd2b88f183f44ab607d43b3d33ea48d390fR3805 anymore

Reply:
oh I see

Reply:
but it's still better to have a single error for 2 part name and 3 part name.

Reply:
Can we update the error class name as well?

Reply:
Sure so I 
1. Updated the error class name and config key to reflect the multiple name parts.
2. use `parseMultiPartIdentifier` to catch more than 1 part.
3. fix test cases.